bokeh/sphinxext/bokeh_gallery.py

```diff
@@ -98,19 +98,25 @@ def config_inited_handler(app, config):
     # left at the end (and remove it in that case)
     extras = set(os.listdir(gallery_dir))
 
-    # app.env.note_dependency(specpath)
-    spec = json.load(open(gallery_file))
-    details = spec["details"]
-
-    names = {x["name"] for x in details}
-    if len(names) < len(details):
-        raise SphinxError(f"gallery file {gallery_file!r} has duplicate names")
+    basepath = app.srcdir.replace('/sphinx/source', '')
```

Comment:
We should compute the "top" path in `bokeh.sphinxext.utils` and make available from there. I think `bokeh-roles` does something like:
```
TOP_PATH = abspath(join(os.curdir, pardir))
```
which is more robust that string replacement.  But ideally I think would would use newer pathlib for this more like
```
TOP_PATH = Path(__file__).resolve().parent.parent.parent
```
In any case we should avoid computing the same path in multiple places in different ways.